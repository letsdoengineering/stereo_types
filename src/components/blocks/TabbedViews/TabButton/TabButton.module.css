.tab-button {
    width: 100%;
    height: 3rem;
    font-weight: 700;
    font-size: 1rem;
    background-color: white;
    border: none;
    border-right: 1px solid var(--outline-gray);
    cursor: pointer;
    transition: 0.5s;
}

.highlighted {
    background-color: var(--highlight-blue-20);
}
