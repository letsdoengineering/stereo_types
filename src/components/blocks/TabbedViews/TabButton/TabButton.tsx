import React from 'react'
import getClassNames from '../../../../utils/get-class-names'
import Text from '../../../basics/Text/Text'
import * as tabButtonStyles from './TabButton.module.css'

type TabButtonProps = {
  isHighlighted?: boolean
  className?: string
  children: React.ReactNode
}

const TabButton: React.FC<TabButtonProps & React.AllHTMLAttributes<HTMLButtonElement>> = ({
  isHighlighted = false,
  className,
  children,
  ...rest
}: TabButtonProps) => {
  const classNames = getClassNames({
    defaultClasses: [tabButtonStyles.tabButton],
    className,
    conditionalClasses: {
      [tabButtonStyles.highlighted]: isHighlighted,
    },
  })

  return (
    <button type='button' className={classNames} {...rest}>
      <Text>{children}</Text>
    </button>
  )
}

export default TabButton
