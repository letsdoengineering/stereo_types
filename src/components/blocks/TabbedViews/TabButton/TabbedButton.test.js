import * as React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { axe } from 'jest-axe'
import TabButton from './TabButton'

describe('TabButton', () => {
    it('has not changed the JSX layout of the page from previous snapshot', () => {
        const tree = renderer.create(<TabButton buttonText='some button' />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('renders the page DOM that is accessible', async () => {
        const { container } = render(<TabButton>some text</TabButton>)
        const results = await axe(container)

        expect(results).toHaveNoViolations()
    })
})
