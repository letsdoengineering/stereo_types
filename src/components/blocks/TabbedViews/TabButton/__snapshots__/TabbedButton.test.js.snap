// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton has not changed the JSX layout of the page from previous snapshot 1`] = `
<button
  buttonText="some button"
  className="tabButton"
  type="button"
>
  <span
    className="sizeS weightNormal textGray"
  />
</button>
`;
